service/history/workflow/update/registry_test.go

```diff
@@ -283,6 +287,68 @@ func TestMessageGathering(t *testing.T) {
 	}
 }
 
+func TestInFlightLimit(t *testing.T) {
+	t.Parallel()
+	limit := 1
+	reg := update.NewRegistry(emptyUpdateStore, update.WithInFlightLimit(
+		func() int { return limit },
+	))
+	upd1, existed, err := reg.FindOrCreate(context.TODO(), "update1")
+	require.NoError(t, err)
+	require.False(t, existed)
+	require.Equal(t, 1, reg.Len())
+
+	t.Run("exceed limit", func(t *testing.T) {
+		_, _, err = reg.FindOrCreate(context.TODO(), "update2")
+		var resExh *serviceerror.ResourceExhausted
+		require.ErrorAs(t, err, &resExh)
+		require.Equal(t, 1, reg.Len())
+	})
+
+	// complete update1 so that it is removed from the registry
+	evStore := mockEventStore{Controller: effect.Immediate(context.TODO())}
+	req := updatepb.Request{
+		Meta:  &updatepb.Meta{UpdateId: "update1"},
+		Input: &updatepb.Input{Name: "not_empty"},
+	}
+	rej := updatepb.Rejection{
+		RejectedRequestMessageId: "update1/request",
+		RejectedRequest:          &req,
+		Failure: &failurepb.Failure{
+			Message: "intentional failure in " + t.Name(),
+		},
+	}
+	require.NoError(t, upd1.OnMessage(context.TODO(), &req, evStore))
+	require.NoError(t, upd1.OnMessage(context.TODO(), &rej, evStore))
+	require.Equal(t, 0, reg.Len(),
+		"completed update should have been removed from registry")
+
+	t.Run("admit next after returning below limit", func(t *testing.T) {
+		_, existed, err = reg.FindOrCreate(context.TODO(), "update2")
+		require.NoError(t, err,
+			"second update should be admitted after first completed")
+		require.False(t, existed)
+	})
+
+	limit = 2 // increase to limit should be seen by UpdateRegistry
```

Comment:
nit: update.Registry will pick up on this when ...

Reply:
eh, I think passive voice is fine in comments. But this reminds me that I'm trying to use t.Log instead of comments in unit tests.

Reply:
Oh, I mainly meant that this was for the update.Registry object, not the UpdateRegistry fn